File: pending/src/main/java/io/mendirl/demo/pending/entity/TransactionEntity.java
package io.mendirl.demo.pending.entity;

import jakarta.persistence.*;
import lombok.*;

import static jakarta.persistence.CascadeType.PERSIST;
import static jakarta.persistence.GenerationType.IDENTITY;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class TransactionEntity {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    private Long id;
    @Version
    private Short version;
    private String name;
    @ToString.Exclude
    @ManyToOne(cascade = {PERSIST})
    private PendingEntity pending;
    @ManyToOne(cascade = {PERSIST})
    private TxCoreEntity txCore;

}

File: kpending/src/main/kotlin/io/mendirl/demo/pending/entity/TransactionEntity.kt
package io.mendirl.demo.pending.entity

import jakarta.persistence.*
import jakarta.persistence.CascadeType.PERSIST
import jakarta.persistence.GenerationType.IDENTITY

@Entity
class TransactionEntity(
        val name: String,
        @ManyToOne(cascade = [PERSIST])
        var pending: PendingEntity? = null,
        @ManyToOne(cascade = [PERSIST])
        var txCore: TxCoreEntity? = null,
        @Id
        @GeneratedValue(strategy = IDENTITY)
        val id: Long? = null,
        @Version
        val version: Short? = null
) {

}
